masu/database/provider_status_accessor.py

```diff
@@ -0,0 +1,212 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Provider status accessors."""
+
+import enum
+import random
+from datetime import datetime, timedelta, timezone
+
+from masu.database.koku_database_access import KokuDBAccess
+from masu.database.provider_db_accessor import ProviderDBAccessor
+from masu.exceptions import MasuProviderError
+
+
+class ProviderStatusCode(enum.IntEnum):
+    """Enum class of provider status.
+
+    This mirrors the model states in koku.api.provider.models
+
+    """
+
+    NEW = 0
+    READY = 1
+    WARNING = 33
```

Comment:
How would eliminating the `WARNING` state facilitate disabling a broken provider?

While you're not entirely incorrect that we could branch our behavior on the retry count, that's a forest-for-the-trees kind of solution - you're missing the bigger picture on how that one state is a part of the whole collection.

It makes no sense to have `READY` and `DISABLED` be states, but then to concoct some alternative logic for the intermediate status that's in between `READY` and `DISABLED`.

Rather than picking apart every individual state to decide whether each one is "necessary" in isolation, I'd like you to try to look at the whole and decide whether we can track at least 3-5 discrete statuses without the concept of a list of states. If you can show me a solution where we don't need the ProviderStatusCode enum at all, then we can talk. Otherwise, I don't see how this is constructive - two inconsistent ways of evaluating state is not an improvement on one consistent mechanism.